### PR TITLE
[GLUTEN-4483][CH]Improve divide

### PR DESCRIPTION
**Pull request by KevinyhZou**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#4483)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

性能测试，通过开发完成的 benchmark_spark_divide_function.cpp 测试 新的divide 函数， 对比CH 的divide函数，以及过去使用FunctionParser的divide 的性能差异，如下
```
2024-01-22 16:51:29.252 <Warning> SignalHandler: LD_PRELOAD is not set, SignalHandler is disabled
2024-01-22T16:51:29+08:00
Running ./benchmark_local_engine
Run on (32 X 2100 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x16)
  L1 Instruction 32 KiB (x16)
  L2 Unified 1024 KiB (x16)
  L3 Unified 11264 KiB (x2)
Load Average: 13.31, 24.04, 42.68
--------------------------------------------------------------------------------------
Benchmark                                            Time             CPU   Iterations
--------------------------------------------------------------------------------------
BM_CHDivideFunction/iterations:10                  268 ms          260 ms           10
BM_SparkDivideFunction/iterations:10               243 ms          243 ms           10
BM_GlutenDivideFunctionParser/iterations:10        322 ms          322 ms           10
```

可见， 新的`SparkDivideFunction` 相对于CH 的`Divide` 大约有10% 左右的提升，相对于Gluten 的`FunctionParser` 的实现约有30% ~ 40% 左右的提升

### CODE REVIEW

File: cpp-ch/local-engine/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +63,7 @@ include_directories(
         ${CMAKE_CURRENT_BINARY_DIR}/proto
         ${THRIFT_INCLUDE_DIR}
         ${CMAKE_BINARY_DIR}/contrib/thrift-cmake
+        ${CMAKE_BINARY_DIR}/contrib/llvm-project/llvm/include
```
Comment (zhanglistar) [inline cpp-ch/local-engine/CMakeLists.txt:66]: why this?

File: cpp-ch/local-engine/CMakeLists.txt [hunk 2]
```diff
@@ -72,6 +73,8 @@ include_directories(
         ${ClickHouse_SOURCE_DIR}/contrib/azure/sdk/storage/azure-storage-blobs/inc
         ${ClickHouse_SOURCE_DIR}/contrib/azure/sdk/core/azure-core/inc
         ${ClickHouse_SOURCE_DIR}/contrib/azure/sdk/storage/azure-storage-common/inc
+        ${ClickHouse_SOURCE_DIR}/contrib/llvm-project/llvm/include
```
Comment (zhanglistar) [inline cpp-ch/local-engine/CMakeLists.txt:76]: why this?

File: cpp-ch/local-engine/Functions/SparkFunctionDivide.h [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                auto null_map_col = ColumnVector<UInt8>::create(vec1->size(), 0);
+                PaddedPODArray<T> & data = vec3->getData();
+                PaddedPODArray<UInt8> & null_map = null_map_col->getData();
+                for (size_t i = 0; i < vec1->size(); ++i)
+                {
+                    L l = vec1->getElement(i);
+                    R r = vec2->getElement(i);
+                    if (r == 0)
+                    {
+                        data[i] = 0;
```
Comment (zhanglistar) [inline cpp-ch/local-engine/Functions/SparkFunctionDivide.h:81]: no need

File: cpp-ch/local-engine/CMakeLists.txt (hunk 1 above)
Comment (KevinyhZou) [reply cpp-ch/local-engine/CMakeLists.txt:66]: if not add this, while compile code, will has the excpeiton 'llvm/IR/IRBuilder.h' file can not be found, as this is defined in `FunctionBinaryArithmetic.h`.

File: cpp-ch/local-engine/CMakeLists.txt (hunk 2 above)
Comment (KevinyhZou) [reply cpp-ch/local-engine/CMakeLists.txt:76]: as above mentioned

File: cpp-ch/local-engine/Functions/SparkFunctionDivide.h (hunk 3 above)
Comment (KevinyhZou) [reply cpp-ch/local-engine/Functions/SparkFunctionDivide.h]: done

**Comment by KevinyhZou** [comment]
reopen

**Pull request closed without merging**